test/test-parse-function.cpp

```diff
@@ -610,13 +622,121 @@ TEST(test_parse, function_statement_without_parameter_list_or_body) {
+    EXPECT_TRUE(p.parse_and_visit_statement(v));
+    EXPECT_THAT(v.visits, ElementsAre("visit_variable_declaration",   // f
+                                      "visit_enter_function_scope",   //
+                                      "visit_variable_declaration",   // x
+                                      "visit_exit_function_scope"));  //
+    EXPECT_THAT(v.errors,
+                ElementsAre(ERROR_TYPE_FIELD(
+                    error_missing_function_body, function_name,
+                    offsets_matcher(&code, strlen(u8"function "), u8"f"))));
+  }
+
+  {
+    padded_string code(u8"class f { m() }"_sv);
+    spy_visitor v;
+    parser p(&code, &v);
+    EXPECT_TRUE(p.parse_and_visit_statement(v));
+    EXPECT_THAT(v.visits, ElementsAre("visit_variable_declaration",  // f
+                                      "visit_enter_class_scope",     //
+                                      "visit_property_declaration",  // m
+                                      "visit_enter_function_scope",  //
+                                      "visit_exit_function_scope",   //
+                                      "visit_exit_class_scope"));    //
+    EXPECT_THAT(v.errors,
+                ElementsAre(ERROR_TYPE_FIELD(
+                    error_missing_function_body, function_name,
+                    offsets_matcher(&code, strlen(u8"class f { "), u8"m"))));
+  }
+
+  {
+    padded_string code(u8"class f { m(x) }"_sv);
+    spy_visitor v;
+    parser p(&code, &v);
+    EXPECT_TRUE(p.parse_and_visit_statement(v));
+    EXPECT_THAT(v.visits, ElementsAre("visit_variable_declaration",  // f
+                                      "visit_enter_class_scope",     //
+                                      "visit_property_declaration",  // m
+                                      "visit_enter_function_scope",  //
+                                      "visit_variable_declaration",  // x
+                                      "visit_exit_function_scope",   //
+                                      "visit_exit_class_scope"));    //
+    EXPECT_THAT(v.errors,
+                ElementsAre(ERROR_TYPE_FIELD(
+                    error_missing_function_body, function_name,
+                    offsets_matcher(&code, strlen(u8"class f { "), u8"m"))));
+  }
+
+  {
+    padded_string code(u8"export default function f()"_sv);
+    spy_visitor v;
+    parser p(&code, &v);
+    EXPECT_TRUE(p.parse_and_visit_statement(v));
+    EXPECT_THAT(v.visits, ElementsAre("visit_variable_declaration",   // f
+                                      "visit_enter_function_scope",   //
+                                      "visit_exit_function_scope"));  //
+    EXPECT_THAT(v.errors,
+                ElementsAre(ERROR_TYPE_FIELD(
+                    error_missing_function_body, function_name,
+                    offsets_matcher(&code, strlen(u8"export default function "),
+                                    u8"f"))));
+  }
+
+  {
+    padded_string code(u8"(function f())"_sv);
```

Comment:
Nit: This is a function expression, not a function statement. The `TEST`'s name implies only function statements are tested. Perhaps the `TEST` should be renamed, or this test case moved into its own `TEST`.